pkg/test/context.go

```diff
@@ -67,15 +81,47 @@ func (f *Framework) newTestCtx(t *testing.T) *TestCtx {
 	}
 }
 
-func NewTestCtx(t *testing.T) *TestCtx {
+func (f *Framework) newContext(t *testing.T) *Context {
+	// TestCtx is used among others for namespace names where '/' is forbidden and must be 63 characters or less
+	id := "osdk-e2e-" + uuid.New()
+
+	var namespace string
+	if f.singleNamespaceMode {
+		namespace = f.Namespace
+	}
+	return &Context{
+		id:                id,
+		t:                 t,
+		namespace:         namespace,
+		namespacedManPath: *f.NamespacedManPath,
+		client:            f.Client,
+		kubeclient:        f.KubeClient,
+		restMapper:        f.restMapper,
+	}
+}
+
+// Deprecated: NewTestCtx exists for historical compatibility
+// You should use NewContext instead of.
+func NewTestCtx(t *testing.T) *TestCtx { //nolint:staticcheck
+	// todo(camilamacedo86): Remove the following line just added for we are able to deprecated TestCtx
+	// need to be done before: 1.0.0
+
 	return Global.newTestCtx(t)
```

Comment:
Does the following change work?

```suggestion
	return NewContext(t)
```